app/controllers/home_controller.rb

```diff
@@ -2,6 +2,14 @@ class HomeController < ApplicationController
 
   skip_before_action :authorize, only: [:index]
 
-  def index; end
+  def index
+    if params[:search]
+      @deals = Deal.includes(:category).filter(params[:category][:category_id]).search(params[:search]).order('created_at DESC')
```

Comment:
this is not the query format that we discussed.
What is the point of discussion if you are not sticking to it? If you had any doubt you should have confirmed with me first. Pls explain